src/charm.py

```diff
@@ -223,13 +271,134 @@ def _on_get_rule_action(self, event: ActionEvent) -> None:
         event.set_results(rule)
 
-    def _configure_auth_proxy_provider(self, event: HookEvent) -> None:
-        """Create access rules and update config."""
-        pass
+    def _on_auth_proxy_config_changed(self, event: AuthProxyConfigChangedEvent) -> None:
+        if not self._oathkeeper_service_is_running:
+            self.unit.status = WaitingStatus("Waiting for Oathkeeper service")
+            event.defer()
+            return
+
+        if not self._peers:
+            self.unit.status = WaitingStatus("Waiting for peer relation")
+            event.defer()
+            return
+
+        access_rules_locations = list()
+        for rule_type in ("allow", "deny"):
+            rules = self._render_access_rules(
+                rule_type=rule_type,
+                protected_urls=event.protected_urls,
+                allowed_endpoints=event.allowed_endpoints,
+                relation_app_name=event.relation_app_name,
+            )
+
+            if rules:
+                # Push the rules to the container
+                filename = f"{self._oathkeeper_access_rules_path}/access-rules-{event.relation_app_name}-{rule_type}.json"
+                self._container.push(filename, str(rules), make_dirs=True)
+                access_rules_locations.append(filename)
+
+        self._set_auth_proxy_relation_peer_data(
+            event.relation_id, dict(access_rules_locations=access_rules_locations)
+        )
+
+        # TODO: Render with all relations' locations
+        self._update_oathkeeper_config_with_access_rules_locations(access_rules_locations)
+
+    def _rule_template(
+        self, rule_id: str, url: str, authenticator: str, mutator: str, error_handler: str
+    ) -> Dict:
+        return {
+            "id": rule_id,
+            "match": {"url": url, "methods": ["GET", "POST"]},
+            "authenticators": [{"handler": authenticator}],
+            "mutators": [{"handler": mutator}],
+            "authorizer": {"handler": "allow"},
+            "errors": [{"handler": error_handler}],
+        }
+
+    def _render_access_rules(
+        self,
+        rule_type: str,
+        protected_urls: List[str],
+        allowed_endpoints: List[str],
+        relation_app_name: str,
+    ) -> Optional[List[dict]]:
+        """Render access rules from a template."""
+        rules = list()
+
+        for url_index, url in enumerate(protected_urls):
+            if url.startswith("https"):
+                # Replace with regex to match both http and https
+                url = url.replace("https", "<https|http>")
+
+            if rule_type == "allow":
+                if not allowed_endpoints:
+                    return None
+                for endpoint in allowed_endpoints:
+                    allow_rule = self._rule_template(
+                        rule_id=f"{relation_app_name}:{endpoint}:{url_index}:allow",
+                        url=f"{url}/{endpoint}",
+                        authenticator="noop",
```

Comment:
shouldn't this be `cookie_session`?

Reply:
Allow rules use `noop`, so that the request for the given endpoint is allowed without having to log in.
Deny rules use `cookie_session`.

Reply:
i thought that if something is going thru oathkeeper we want it to be authenticated, the decision endpoint is to take away that logic of checking the header in the client application

i might be completely wrong but, in the case of using the `noop` for `allow` rules we are basically doing a proxy pass, which albeit might be a usecase should probably not be the default one